multiversx_sdk_rust_contract_builder/builder.py

```diff
@@ -39,16 +39,17 @@ def build_project(
     ensure_output_folder_is_empty(parent_output_folder)
 
     outcome = BuildOutcome(metadata, options)
-    contracts_folders = get_contracts_folders(project_folder)
+    contracts = get_contracts(project_folder)
```

Comment:
Perhaps keep this as it were, then add a check below it:

```
ensure_distinct_contract_names(contracts_folders)
```

Which makes use of `get_contract_name_and_version`.

The validation logic would be a bit more explicit this way.